packages/wonder-blocks-button/types.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import React from "react";
+import type {Element} from "react";
+
+import type {ValidTints} from "wonder-blocks-color";
+
+export type SharedProps = {
+    /**
+     * Text to appear on the button.
+     */
+    children: string,
+
+    /**
+     * An icon, displayed to the left of the title.
+     */
+    icon?: string,
+
+    /**
+     * If true, replaces the contents with a spinner.
+     */
+    // TODO(yejia): Implement once spinner is implemented.
+    // spinner: boolean,
+
+    /**
+     * The color of the button, either blue (default) or red.
+     */
+    color?: ValidTints,
```

Comment:
It feels like a number of these should be required and then we should set sensible defaults using `static defaultProps`. Specifically color (Color.blue), kind (primary), light (false), size (default), and disabled (false). If you do this inside the Button component then you won't have to inside ButtonCore, you can just use the values.